### PR TITLE
Use config.ID for Receivers config 

### PR DESCRIPTION
**Pull request by bogdandrutu**
Depends on #2869

### CODE REVIEW

File: cmd/schemagen/schemagen/cli_all.go [hunk 1]
```diff
@@ -27,8 +26,8 @@ func createAllSchemaFiles(components component.Factories, env env) {
 	}
 }
 
-func getAllConfigs(components component.Factories) []config.NamedEntity {
-	var cfgs []config.NamedEntity
+func getAllConfigs(components component.Factories) []interface{} {
```
Comment (jrcamp) [inline cmd/schemagen/schemagen/cli_all.go:29]: No more common interface?

Comment (bogdandrutu) [reply cmd/schemagen/schemagen/cli_all.go:29]: not yet, I will add one later. Since now I will change the configs one by one.

Comment (jrcamp) [reply cmd/schemagen/schemagen/cli_all.go:29]: what about `identifiable`? I see you didn't export though.

Comment (bogdandrutu) [reply cmd/schemagen/schemagen/cli_all.go:29]: This `cmd` actually does not need any common, but let me see how the config will look like and I will have some common interface.

**Comment by tigrannajaryan** [comment]
Please rebase.

**Comment by tigrannajaryan** [comment]
Needs a rebase.

File: config/identifiable.go [hunk 2]
```diff
@@ -78,6 +91,12 @@ func (id ComponentID) Name() string {
 	return id.nameVal
 }
 
+// WithName returns a new ComponentID, with the same type as the current instance,
+// and the given name.
+func (id ComponentID) WithName(nameVal string) ComponentID {
```
Comment (tigrannajaryan) [inline config/identifiable.go:96]: The usage of `WithName` in the code looks a bit unclear, e.g. `defaultID.WithName` is somewhat weird. How about simply having a `NewIDWthName` func?

File: receiver/jaegerreceiver/config.go [hunk 3]
```diff
@@ -76,9 +76,9 @@ func DefaultServerConfigUDP() ServerConfigUDP {
 
 // Config defines configuration for Jaeger receiver.
 type Config struct {
-	config.ReceiverSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
-	Protocols               `mapstructure:"protocols"`
-	RemoteSampling          *RemoteSamplingConfig `mapstructure:"remote_sampling"`
+	*config.ReceiverSettings `mapstructure:"-"`
```
Comment (tigrannajaryan) [inline receiver/jaegerreceiver/config.go:79]: It is not clear why this is embedded by pointer. It does not see to be optional, is it?

File: config/identifiable.go (hunk 2 above)
Comment (bogdandrutu) [reply config/identifiable.go:96]: That will make us to pass type + name. I can try to see how it looks in couple of cases and we can decide

File: receiver/jaegerreceiver/config.go (hunk 3 above)
Comment (bogdandrutu) [reply receiver/jaegerreceiver/config.go]: Done. Fixed the pointer problem.

**Comment by tigrannajaryan** [comment]
@bogdandrutu is this ready for final review / merging or you are still working on it?

File: config/identifiable.go (hunk 2 above)
Comment (bogdandrutu) [reply config/identifiable.go]: Done.

**Comment by bogdandrutu** [comment]
@tigrannajaryan now is updated and fixed the `WithName`.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**